content/blog/posts/2021-03-11-getting-started-with-serverless.md

```diff
@@ -0,0 +1,174 @@
+
+## What we'll work on today
+
+A common need for teams today in their applications is a GraphQL server as the API for data persistance. While not challenging to setup a basic service for use locally, there tend to often be quite a bit more steps for deployment of these APIs for use in various environments. We'll use the Serverless framework to get us started quickly and give us the ability to deploy to many common cloud providers. 
+
+```bash
+npx serverless
+```
+
+Follow the prompts and this will give you a basic serverless project with all you need to get started. We can change the `serverless.yml` to define our function and the handler specifically for HTTP requests by adding the following:
+
+```yaml
+functions:
+  graphql:
+    handler: handler.graphql
+    events:
+      - http:
+          path: graphql
+          method: post
+```
+
+For our needs, we'll want to also add a package for Graphql and I'm going to use [apollo-server-lambda](https://www.npmjs.com/package/apollo-server-lambda).
+
+```bash
+yarn add apollo-server-lambda
+```
+
+Now let's change our function to create a GraphQL server and use it to respond to query requests. For our demonstration, we'll just have it respond to a string for a simple query. 
+
+```js
+import { ApolloServer, gql } from 'apollo-server-lambda';
+
+const typeDefs = gql`
+  type Query {
+    hello: String
+  }
+`;
+
+const resolvers = {
+  Query: {
+    hello: () => 'Hello world!',
+  }
+};
+
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  introspection: true // this is nice in testing with GraphQL playground
+});
+
+export const graphql = server.createHandler();
+```
+
+## Try this out locally
+
+Now that you have a basic Lambda as your GraphQL server, you probably want to try it out before getting it out in the internet. Not that this is a requirement, most free tiers of providers make this a free experiment and so there's not risk to just pushing and testing. That said, a couple of packages also make it simple to test your function locally. 
+
+For our basic needs, we'll use the [serverless-offline](https://www.npmjs.com/package/serverless-offline) plugin to emulate AWS Lamdba and API Gateway services on a local server. In this regard we are choosing a vendor in a way, but for our purposes of chosing a path and pushing the code somewhere AWS is the one I'm personally most familiar with and that is the stated provider by default when generating a serverless project.
+
+Also, I like to write in module syntax, so adding [serverless-bundle](https://www.npmjs.com/package/serverless-bundle) to handle basic Webpack configuration for me gives me that flexibility. You can use it for adding Typescript, ESLint, Bable transpiling, and to generate source maps. For this use case though, I'll keep it simple with just our friend JavaScript.
```

Comment:
```suggestion
Also, I like to write in module syntax, so adding [serverless-bundle](https://www.npmjs.com/package/serverless-bundle) to handle basic Webpack configuration for me gives me that flexibility. It can be used for adding Typescript, ESLint, Babel transpiling, and to generate source maps. For this use case though, we'll keep it simple with just our friend JavaScript.
```